lib/mustache_context.dart

```diff
@@ -124,12 +124,30 @@ class _MustacheContext extends MustacheToString implements MustacheContext {
   }
 
   dynamic _getActualValue(String key) {
-    try {
-      return ctx[key];
-    } catch (NoSuchMethodError) {
-      //Try to make dart2js understand that when we define USE_MIRRORS = false
-      //we do not want to use any reflector
-      return (useMirrors && USE_MIRRORS) ? ctxReflector[key] : null;
+    //Try to make dart2js understand that when we define USE_MIRRORS = false
+    //we do not want to use any reflector as that inflates the generated
+    //javascript.
+    if (useMirrors && USE_MIRRORS) {
+      if (reflect(ctx).type.instanceMembers.containsKey(new Symbol("[]"))) {
+        MethodMirror m = reflect(ctx).type.instanceMembers[new Symbol("[]")];
+        TypeMirror reflectedString = reflectType(String);
+        if (reflectedString.isAssignableTo(m.parameters[0].type)) {
```

Comment:
It looks like dart2js has some problems with this one:

```
00:26 +8 -10: Deep subcontext test [E]                                                             
  UnimplementedError
  dart:_internal                                      dart<.JsTypeMirror.isAssignableTo
  package:mustache4dart/mustache_context.dart 257:9   dart<._ObjectReflector.fieldValue
  package:mustache4dart/mustache_context.dart 241:26  dart<._ObjectReflector.$index
  package:mustache4dart/mustache_context.dart 123:14  dart<._MustacheContext._getActualValue
  package:mustache4dart/mustache_context.dart 100:13  dart<._MustacheContext._getContextForKey
  package:mustache4dart/mustache_context.dart 69:18   dart<._MustacheContext._getInThisOrParent
  package:mustache4dart/mustache_context.dart 65:12   dart<._MustacheContext.field
  test/mustache_context_test.dart 192:12              dart<.main.<fn>
```

I'm afraid I have to trap some exception after all. Do you mind sharing your usecase? Maybe I can try the exception trapping deeper in order not to hit the performance .